src/datascience-ui/data-explorer/sliceControl.tsx

```diff
@@ -4,7 +4,70 @@ import { IGetSliceRequest } from '../../client/datascience/data-viewing/types';
 
 import './sliceControl.css';
 
-const sliceRegEx = /^\s*(?<StartRange>\d+:)|(?<StopRange>:\d+)|(?:(?<Start>-?\d+)(?::(?<Stop>-?\d+))?(?::(?<Step>-?\d+))?)\s*$/;
+const sliceRegEx = /^\s*((?<StartRange>-?\d+:)|(?<StopRange>-?:\d+)|(?:(?<Start>-?\d+)(?::(?<Stop>-?\d+))?(?::(?<Step>-?\d+))?))\s*$/;
+const textFieldStyles = {
```

Comment:
How come these styles can't go in the sliceControl.css?

Reply:
This gets passed to the FluentUI components (which expect a JSON format keyed by the name of each type of child element in the dropdown) https://developer.microsoft.com/en-us/fluentui#/controls/web/dropdown#IDropdownStyles

Reply:
Maybe a comment saying why? I don't currently see a way around it, but it stinks when we have styles in the code. They're harder to find and correlate with other stuff.